src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -29,4 +31,8 @@ public NakadiKafkaConsumer createNakadiConsumer(final String topic, final List<K
         return new NakadiKafkaConsumer(getConsumer(), topic, kafkaCursors, pollTimeout);
     }
 
+    @Override
+    public void updateProperties(final Properties kafkaProperties) {
+        kafkaProducer = new KafkaProducer<>(kafkaProperties);
```

Comment:
1. The name of the method `createProducer()` is misleading here. It should be called whenever a kafka producer is needed and not be cached or kept by a caller (same as for the consumer). I gonna change the name of the method. But you hit a very good point: I wanted to check if every caller does it right and totally forgot.
2. I agree with @rcillo, I think it's very hypothetical. Changes to the properties should only happen within this class and thus it's the responsibility of the one who does it to make it right and eventually call `notifyPropertiesListeners()`. I would prefer to not overcomplicated it here. And even if somebody else would call `updateProperties()` it shouldn't be an issue since assigning a new kafka producer should be atomic as well a returning the field. However, I evaluate making the `kafkaProducer` field volatile as you suggested in another discussion.
3. Oh yes, I will add this!